pinot-core/src/test/java/org/apache/pinot/core/data/manager/realtime/ConsumptionDelayTrackerTest.java

```diff
@@ -0,0 +1,238 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.core.data.manager.realtime;
+
+import org.apache.pinot.common.metrics.ServerMetrics;
+import org.apache.pinot.spi.metrics.PinotMetricUtils;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class ConsumptionDelayTrackerTest {
+
+  private static final int TIMER_THREAD_TICK_INTERVAL_MS = 100;
+  private static final int INITIAL_TIMER_THREAD_DELAY_MS = 100;
+
+  private void sleepMs(long timeMs) {
```

Comment:
Please avoid sleeps in test. Use ways to modify clock time as needed. I think there are a couple of different examples in the code base for this.

Reply:
+1 to avoiding sleeps in test. 
You can extend `java.time.Clock` to define your own ticking clock or use one of the existing clocks like `FixedClock`, `OffsetClock` etc. There are also other implementations like `org.threeten.extra.MutableClock` which is likely what you need. Manually set the clock to a time to advance the clock.

Reply:
ACK, I will work on this but likely towards the end once all code issues are closed.

Reply:
This has been addressed by commit 9a6b0e5fb705ac00f25b2e41f5ca7690b05ca34e